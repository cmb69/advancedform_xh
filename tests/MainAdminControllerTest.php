<?php

namespace Advancedform;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Plib\CsrfProtector;
use Plib\View;

class MainAdminControllerTest extends TestCase
{
    public function testRendersFormsOverview(): void
    {
        global $tx;
        $tx = XH_includeVar("../../cmsimple/languages/en.php", "tx");
        $formGateway = $this->createStub(FormGateway::class);
        $formGateway->method("findAll")->willReturn(["Contact" => $this->form()]);
        $csrfProtector = $this->createStub(CsrfProtector::class);
        $csrfProtector->method("token")->willReturn("0123456789ABCDEF");
        $sut = new MainAdminController(
            $formGateway,
            "./",
            XH_includeVar("./config/config.php", "plugin_cf")["advancedform"],
            XH_includeVar("./languages/en.php", "plugin_tx")["advancedform"],
            $csrfProtector,
            new View("./templates/", XH_includeVar("./languages/en.php", "plugin_tx")["advancedform"])
        );
        Approvals::verifyHtml($sut->formsAdministrationAction());
    }

    private function form(): Form
    {
        return Form::createFromArray([
            "captcha" => true,
            "name" => "Contact",
            "title" => "Contact",
            "to_name" => "Webmaster",
            "to" => "webmaster@example.com",
            "cc" => "",
            "bcc" => "",
            "thanks_page" => "",
            "store" => false,
            "fields" => [
                [
                    "field" => "Name",
                    "label" => "Name",
                    "type" => "from_name",
                    "props" => "¦¦¦",
                    "required" => true
                ],
                [
                    "field" => "E_Mail",
                    "label" => "E-Mail",
                    "type" => "from",
                    "props" => "¦¦¦",
                    "required" => true
                ],
                [
                    "field" => "Phone",
                    "label" => "Phone",
                    "type" => "custom",
                    "props" => "¦¦¦/^[0-9\\/\\-\\+\\ \\(\\)]*$/¦Field '%s' doesn't contain a valid phone number!",
                    "required" => false
                ],
                [
                    "field" => "Comment",
                    "label" => "Comment",
                    "type" => "textarea",
                    "props" => "¦¦¦",
                    "required" => true
                ],
            ],
        ]);
    }
}
