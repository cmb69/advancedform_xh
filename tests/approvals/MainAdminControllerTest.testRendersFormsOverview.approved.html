<!-- advancedform/forms-admin -->
<div id="advfrm-form-list">
  <h1>Advancedform â€“ Mail Forms</h1>
  <form class="advfrm-add-form" action="./?advancedform&amp;admin=plugin_main&amp;action=new" method="post" >
    <button title="Add"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512" width="1em" height="1em" fill="currentColor"><!--!Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free Copyright 2025 Fonticons, Inc.--><path d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 144L48 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l144 0 0 144c0 17.7 14.3 32 32 32s32-14.3 32-32l0-144 144 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-144 0 0-144z"/></svg></button>
    <input type="hidden" name="advancedform_token" value="0123456789ABCDEF">
  </form>
  <form class="advfrm-import-form" action="./?advancedform&amp;admin=plugin_main&amp;action=import&amp;form=" method="post" >
    <button title="Import"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" width="1em" height="1em" fill="currentColor"><!--!Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free Copyright 2025 Fonticons, Inc.--><path d="M288 109.3L288 352c0 17.7-14.3 32-32 32s-32-14.3-32-32l0-242.7-73.4 73.4c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3l128-128c12.5-12.5 32.8-12.5 45.3 0l128 128c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L288 109.3zM64 352l128 0c0 35.3 28.7 64 64 64s64-28.7 64-64l128 0c35.3 0 64 28.7 64 64l0 32c0 35.3-28.7 64-64 64L64 512c-35.3 0-64-28.7-64-64l0-32c0-35.3 28.7-64 64-64zM432 456a24 24 0 1 0 0-48 24 24 0 1 0 0 48z"/></svg></button>
    <input type="hidden" name="advancedform_token" value="0123456789ABCDEF">
  </form>
  <table>
    <tr>
      <td class="tool">
        <form class="advfrm-delete-form" action="./?advancedform&amp;admin=plugin_main&amp;action=delete&amp;form=Contact" method="post" onsubmit="return confirm('The form and all it\'s information will be deleted!')">
          <button title="Delete"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512" width="1em" height="1em" fill="currentColor"><!--!Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free Copyright 2025 Fonticons, Inc.--><path d="M135.2 17.7L128 32 32 32C14.3 32 0 46.3 0 64S14.3 96 32 96l384 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-96 0-7.2-14.3C307.4 6.8 296.3 0 284.2 0L163.8 0c-12.1 0-23.2 6.8-28.6 17.7zM416 128L32 128 53.2 467c1.6 25.3 22.6 45 47.9 45l245.8 0c25.3 0 46.3-19.7 47.9-45L416 128z"/></svg></button>
          <input type="hidden" name="advancedform_token" value="0123456789ABCDEF">
        </form>
      </td>
      <td class="tool">
        <form class="advfrm-template-form" action="./?advancedform&amp;admin=plugin_main&amp;action=template&amp;form=Contact" method="post" onsubmit="return confirm('A possibly existing Contact.tpl and Contact.css file will be overwritten!')">
          <button title="Create template"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 384 512" width="1em" height="1em" fill="currentColor"><!--!Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free Copyright 2025 Fonticons, Inc.--><path d="M64 464c-8.8 0-16-7.2-16-16L48 64c0-8.8 7.2-16 16-16l160 0 0 80c0 17.7 14.3 32 32 32l80 0 0 288c0 8.8-7.2 16-16 16L64 464zM64 0C28.7 0 0 28.7 0 64L0 448c0 35.3 28.7 64 64 64l256 0c35.3 0 64-28.7 64-64l0-293.5c0-17-6.7-33.3-18.7-45.3L274.7 18.7C262.7 6.7 246.5 0 229.5 0L64 0zm56 256c-13.3 0-24 10.7-24 24s10.7 24 24 24l144 0c13.3 0 24-10.7 24-24s-10.7-24-24-24l-144 0zm0 96c-13.3 0-24 10.7-24 24s10.7 24 24 24l144 0c13.3 0 24-10.7 24-24s-10.7-24-24-24l-144 0z"/></svg></button>
          <input type="hidden" name="advancedform_token" value="0123456789ABCDEF">
        </form>
      </td>
      <td class="tool">
        <form class="advfrm-copy-form" action="./?advancedform&amp;admin=plugin_main&amp;action=copy&amp;form=Contact" method="post" >
          <button title="Copy"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" width="1em" height="1em" fill="currentColor"><!--!Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free Copyright 2025 Fonticons, Inc.--><path d="M288 448L64 448l0-224 64 0 0-64-64 0c-35.3 0-64 28.7-64 64L0 448c0 35.3 28.7 64 64 64l224 0c35.3 0 64-28.7 64-64l0-64-64 0 0 64zm-64-96l224 0c35.3 0 64-28.7 64-64l0-224c0-35.3-28.7-64-64-64L224 0c-35.3 0-64 28.7-64 64l0 224c0 35.3 28.7 64 64 64z"/></svg></button>
          <input type="hidden" name="advancedform_token" value="0123456789ABCDEF">
        </form>
      </td>
      <td class="tool">
        <form class="advfrm-export-form" action="./?advancedform&amp;admin=plugin_main&amp;action=export&amp;form=Contact" method="post" onsubmit="return confirm('A possibly existing Contact.json file will be overwritten!')">
          <button title="Export"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" width="1em" height="1em" fill="currentColor"><!--!Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free Copyright 2025 Fonticons, Inc.--><path d="M288 32c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 242.7-73.4-73.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l128 128c12.5 12.5 32.8 12.5 45.3 0l128-128c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L288 274.7 288 32zM64 352c-35.3 0-64 28.7-64 64l0 32c0 35.3 28.7 64 64 64l384 0c35.3 0 64-28.7 64-64l0-32c0-35.3-28.7-64-64-64l-101.5 0-45.3 45.3c-25 25-65.5 25-90.5 0L165.5 352 64 352zm368 56a24 24 0 1 1 0 48 24 24 0 1 1 0-48z"/></svg></button>
          <input type="hidden" name="advancedform_token" value="0123456789ABCDEF">
        </form>
      </td>
      <td class="name"><a href="./?advancedform&amp;admin=plugin_main&amp;action=edit&amp;form=Contact" title="Edit">Contact</a></td>
      <td class="script" title="Script code to insert on a page">
        <input type="text" readonly onclick="this.select()" value="{{{advancedform('Contact')}}}"></input>
      </td>
    </tr>
  </table>
</div>
